compiler/parser.jou

```diff
@@ -212,11 +155,115 @@ def decorate(stmt: AstStatement*, decor: byte*, location: Location) -> None:
         fail(location, msg)
 
 
+enum WhatAreWeParsing:
+    TopLevel  # must be first, so that this is the default (zero)
+    ClassBody
+    FunctionBody
+    MethodBody
+
+
+# Errors if e.g. trying to define a class inside a function (other way around it would become a method)
+def check_statement_not_misplaced(
+    stmt_kind: AstStatementKind,
+    what_are_we_parsing: WhatAreWeParsing,
+    location: Location,
+) -> None:
+    msg: byte[500]
+
+    match what_are_we_parsing:
+        case WhatAreWeParsing.TopLevel:
+            inside_xyz = "<internal error>"  # hopefully nobody sees this....
+        case WhatAreWeParsing.ClassBody:
+            inside_xyz = "inside a class"
+        case WhatAreWeParsing.FunctionBody:
+            inside_xyz = "inside a function"
+        case WhatAreWeParsing.MethodBody:
+            inside_xyz = "inside a method"
+
+    match stmt_kind:
+        case AstStatementKind.If:
+            # allowed everywhere, so you can do e.g. "if WINDOWS:" however you want
+            pass
+
+        case AstStatementKind.Import:
+            if what_are_we_parsing != WhatAreWeParsing.TopLevel:
+                fail(location, "imports must be in the beginning of the file")
+
+        case (
+            AstStatementKind.ExpressionStatement
+            | AstStatementKind.Assert
+            | AstStatementKind.Pass
+            | AstStatementKind.Return
+            | AstStatementKind.Match
+            | AstStatementKind.WhileLoop
+            | AstStatementKind.ForLoop
+            | AstStatementKind.Break
+            | AstStatementKind.Continue
+            | AstStatementKind.DeclareLocalVar
+            | AstStatementKind.Assign
+            | AstStatementKind.InPlaceAdd
+            | AstStatementKind.InPlaceSub
+            | AstStatementKind.InPlaceMul
+            | AstStatementKind.InPlaceDiv
+            | AstStatementKind.InPlaceMod
+        ):
+            match what_are_we_parsing:
+                case WhatAreWeParsing.TopLevel:
+                    fail(location, "you must put your code inside a function")
```

Comment:
This error message is not nice if the problem is forgotten `global`.

For example, with `x: int` this in `foo.jou`, I get:

```
compiler error in file "foo.jou", line 1: you must put your code inside a function
```

Ideally the error would be e.g. `'global' keyword needed to create a global variable`